### PR TITLE
replace variable in key

### LINKED ISSUE #844: 'subscriptionId' and 'resourceGroupName' in object key are not replaced
image

### PR DESCRIPTION
**Pull request by jordan13game**
resolves https://github.com/Azure/oav/issues/844

### CODE REVIEW

File: test/apiScenario/fixtures/specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/scenarios/testKeyReplace.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+scope: ResourceGroup
+
+scenarios:
+  - scenario: StorageBasicExample
+    steps:
+      - step: createStorageAccount
+        exampleFile: ../examples/StorageAccountCreateUserAssignedIdentityWithFederatedIdentityClientId.json
+        requestUpdate:
+          - move: /parameters/identity/userAssignedIdentities/~1subscriptions~1$(subscriptionId)~1resourceGroups~1$(resourceGroupName)~1providers~1Microsoft.ManagedIdentity~1userAssignedIdentities~1{managed-identity-name}
+            from: /parameters/identity/userAssignedIdentities/~1subscriptions~1{subscription-id}~1resourceGroups~1res9101~1providers~1Microsoft.ManagedIdentity~1userAssignedIdentities~1{managed-identity-name}
```
Comment (leniatgh) [inline test/apiScenario/fixtures/specification/storage/resource-manager/Microsoft.Storage/stable/2021-08-01/scenarios/testKeyReplace.yaml]: can we add one more test case using `replace`?

File: lib/apiScenario/gen/restlerApiScenarioGenerator.ts [hunk 2]
```diff
@@ -411,7 +411,8 @@ export class RestlerApiScenarioGenerator {
         continue;
       }
       for (const method of Object.keys(dependencies[path])) {
-        const operationId = this.getOperationId(path, method);
+        const operationId =
+          this.getOperationId(path, method) || this.getOperationId(path + "/", method);
```
Comment (leniatgh) [inline lib/apiScenario/gen/restlerApiScenarioGenerator.ts:416]: what's this for?

File: lib/apiScenario/newmanReportValidator.ts [hunk 3]
```diff
@@ -205,6 +205,9 @@ export class NewmanReportValidator {
         }
 
         it.assertions.forEach((assertion) => {
+          if (assertion.test.includes("response status code assertion")) {
```
Comment (leniatgh) [inline lib/apiScenario/newmanReportValidator.ts]: we can define a constant string in PostmanHelper, for better maintainability

Comment (leniatgh) [reply lib/apiScenario/newmanReportValidator.ts]: It seems some assertions are implemented under the response status code assertion, we should take those out

**Review by leniatgh — COMMENTED** [review]
Looks good, some minor comments

File: lib/apiScenario/gen/restlerApiScenarioGenerator.ts (hunk 2 above)
Comment (jordan13game) [reply lib/apiScenario/gen/restlerApiScenarioGenerator.ts:416]: some paths end with "/", but RESTler will remove it in dependencies. So we can't find it in swagger

**Review by leniatgh — APPROVED** [review]

**Pull request merged**